controllers/bootstrap/init.go

```diff
@@ -6,36 +6,54 @@ import (
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
 
+	"k8s.io/client-go/rest"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/medik8s/node-healthcheck-operator/controllers/console"
 	"github.com/medik8s/node-healthcheck-operator/controllers/defaults"
 	"github.com/medik8s/node-healthcheck-operator/controllers/rbac"
 	"github.com/medik8s/node-healthcheck-operator/controllers/utils"
 )
 
-// Initialize runs some bootstrapping code:
+// Initializer runs some bootstrapping code:
 // - setup role aggregation
 // - create default NHC
 // - create console plugin
-func Initialize(ctx context.Context, mgr ctrl.Manager, log logr.Logger) error {
+type initializer struct {
+	cl     client.Client
+	config *rest.Config
+	logger logr.Logger
+}
+
+// New returns a new Initializer
+func New(mgr ctrl.Manager, logger logr.Logger) *initializer {
```

Comment:
NIT: Can we use more meaningful function name? e.g. `CreateInitializer` or `NewInitializer"

Reply:
- functions for struct initialization typically use `New`, not `Create`
- when there is one such struct + function only, and the package also is just about this single struct, typically it's just `New` without anything else, in order to avoid verbosity
- however, in this case, the package name was bad. Renamed to `initializer`. So the function call now is `initializer.New()`, no need anymore for extending the function name 🙂